client/app/widgets/CarePlanContacts/__tests__/index.spec.jsx

```diff
@@ -17,105 +17,121 @@ const contacts = [
-      ).toEqual('Test1 Lastname');
-      expect(
-        wrapper.find('input[aria-label="Phone number"]').props().defaultValue,
-      ).toEqual(undefined);
-      wrapper.find('input[aria-label="Phone number"]').instance().value = '4160000000';
-      await act(async () => {
-        await wrapper.find('input[type="submit"]').simulate('click');
       });
-      axiosPatchSpy();
-      expect(wrapper.text()).toContain('4160000000');
+      const { container } = render(component);
+
+      const editLink = container.querySelector('a[aria-label="Edit"]');
+      expect(editLink).toBeInTheDocument();
+      expect(screen.queryByText('Edit Contact')).not.toBeInTheDocument();
+
+      userEvent.click(editLink);
+      expect(screen.getByText('Edit Contact')).toBeInTheDocument();
+      expect(screen.getByLabelText('Name')).toHaveDisplayValue(
+        contacts[0].name,
+      );
+      expect(screen.getByLabelText('Phone number')).toHaveDisplayValue('');
+
+      userEvent.type(screen.getByLabelText('Phone number'), newPhoneNumber);
+      expect(screen.getByLabelText('Phone number')).toHaveDisplayValue(
+        newPhoneNumber,
+      );
+
+      userEvent.click(screen.getByText('Submit'));
+      await waitFor(() => expect(screen.getByText(newPhoneNumber)).toBeInTheDocument());
     });
 
     it('opens a modal and does not submit the form successfully', async () => {
-      const axiosPatchSpy = jest
-        .spyOn(axios, 'patch')
-        .mockImplementation(() => Promise.reject(error));
-      const wrapper = mount(component);
-      wrapper.find('a[aria-label="Edit"]').at(0).simulate('click');
-      expect(wrapper.text()).toContain('Edit Contact');
-      expect(
-        wrapper.find('input[aria-label="Name"]').props().defaultValue,
-      ).toEqual('Test1 Lastname');
-      expect(
-        wrapper.find('input[aria-label="Phone number"]').props().defaultValue,
-      ).toEqual(undefined);
-      wrapper.find('input[aria-label="Phone number"]').instance().value = '4160000000';
-      await act(async () => {
-        await wrapper.find('input[type="submit"]').simulate('click');
-      });
-      expect(axiosPatchSpy()).rejects.toEqual(error);
-      expect(wrapper.text()).toContain('Error');
+      const error = new Error('Error');
+      const axiosPostSpy = jest.spyOn(axios, 'patch').mockRejectedValue(error);
+      const { container } = render(component);
+
+      const editLink = container.querySelector('a[aria-label="Edit"]');
+      expect(editLink).toBeInTheDocument();
+      expect(screen.queryByText('Edit Contact')).not.toBeInTheDocument();
+
+      userEvent.click(editLink);
+      expect(screen.getByText('Edit Contact')).toBeInTheDocument();
+      expect(screen.getByLabelText('Name')).toHaveDisplayValue(
+        contacts[0].name,
+      );
+      expect(screen.getByLabelText('Phone number')).toHaveDisplayValue('');
+      expect(screen.queryByText('Error: Error')).not.toBeInTheDocument();
```

Comment:
I think we can remove this since if there was an error, it would throw in the test console and fail the test.

Reply:
All of these are me just being extra cautious and I agreed with you, we can remove those.

Reply:
Yeah no worries! I think they're good tests to write as you're building out tests :)